Kernel/Devices/VMWareSVGADevice.cpp

```diff
@@ -0,0 +1,271 @@
+/*
+ * Copyright (c) 2021, the Serenity OS Developers
+ * Copyright (c) 2021, Alexander Richards <electrodeyt@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "VMWareSVGADevice.h"
+#include <AK/Singleton.h>
+#include <Kernel/PCI/Access.h>
+#include <Kernel/PCI/IDs.h>
+#include <Kernel/VM/AnonymousVMObject.h>
+#include <Kernel/VM/MemoryManager.h>
+#include <LibC/errno_numbers.h>
+#include <LibC/sys/ioctl_numbers.h>
+
+namespace Kernel {
+
+VMWareSVGADevice::VMWareSVGADevice(PCI::Address addr)
+    : BlockDevice(29, 0)
+    , PCI::Device(addr)
+{
+    PCI::Address address = pci_address();
+    // Get the base address of the IO addresses
```

Comment:
The comments are describing what the code does, which is pretty useless as I can read the code right here. Is there something non-obvious being done here?